spec.md

```diff
@@ -2236,8 +2236,9 @@ circuit Foo:
     wire cond: UInt<1>
     wire a: UInt
     wire b: UInt
+    node fd = SInt<32>(-2)
```

Comment:
we basically may need a new type in Firrtl and CIRCT, but anyway it LGTM. in CIRCT we need to check the fd shouldn't be used for circuit construction.